### PR TITLE
CI split between master (full) and PR (smoke) for NRF

### PR DESCRIPTION
**Pull request by andy31415**
#### Problem
We spend a lot of time waiting for PR CI.

#### Change overview
Moved full NRF into master builds only, moved the PR builds as a "smoke test" only that is expected to finish faster (under 30 min).

PR smoke test will only do an app compile. Master pushes will compile all apps and run zephyr unit tests.

#### Testing
CI will validate.

### CODE REVIEW

File: .github/workflows/smoketest-nrfconnect.yaml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                   .environment/pigweed-venv/*.log
+            - name: Update nRF Connect SDK revision to the currently recommended.
+              timeout-minutes: 10
+              run: scripts/run_in_build_env.sh "python3 scripts/setup/nrfconnect/update_ncs.py --update --shallow"
+            - name: Build example nRF Connect SDK Lock App on nRF52840 DK
+              timeout-minutes: 10
+              run: |
+                  scripts/examples/nrfconnect_example.sh lock-app nrf52840dk_nrf52840
+                  .environment/pigweed-venv/bin/python3 scripts/tools/memory/gh_sizes.py \
+                    nrfconnect nrf52840dk_nrf52840 lock-app \
```
Comment (mrjerryjohns) [inline .github/workflows/smoketest-nrfconnect.yaml:68]: Shouldn't this be the all clusters app?

**Review by mrjerryjohns — APPROVED** [review]

File: .github/workflows/smoketest-nrfconnect.yaml [hunk 2]
```diff
@@ -0,0 +1,77 @@
+# Copyright (c) 2020 Project CHIP Authors
```
Comment (bzbarsky-apple) [inline .github/workflows/smoketest-nrfconnect.yaml:1]: See question at https://github.com/project-chip/connectedhomeip/pull/15607#discussion_r815161551

File: .github/workflows/smoketest-nrfconnect.yaml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                  path: |
+                   .environment/gn_out/.ninja_log
+                   .environment/pigweed-venv/*.log
+            - name: Update nRF Connect SDK revision to the currently recommended.
+              timeout-minutes: 10
+              run: scripts/run_in_build_env.sh "python3 scripts/setup/nrfconnect/update_ncs.py --update --shallow"
+            - name: Build example nRF Connect SDK Lock App on nRF52840 DK
+              timeout-minutes: 10
+              run: |
+                  scripts/examples/nrfconnect_example.sh lock-app nrf52840dk_nrf52840
```
Comment (Damian-Nordic) [inline .github/workflows/smoketest-nrfconnect.yaml:66]:
```suggestion
                  scripts/examples/nrfconnect_example.sh all-clusters-app nrf52840dk_nrf52840 -DBUILD_WITH_DFU=1
```
I agree with @mrjerryjohns that all-clusters-app seems the best choice if we want to build once app per platform. Also, let's build it with the DFU support to additionally increase the coverage.

File: .github/workflows/smoketest-nrfconnect.yaml (hunk 1 above)
Comment (andy31415) [reply .github/workflows/smoketest-nrfconnect.yaml:68]: I picked the first app that was compiled and it also seemed we may want to have a mix of smoke tests compiling either all clusters or some other app, to exercise common per-example code.

We can update it to other apps in other PRs if we think that is appropriate. I would leave this choice up to platform maintainers.

File: .github/workflows/smoketest-nrfconnect.yaml (hunk 3 above)
Comment (andy31415) [reply .github/workflows/smoketest-nrfconnect.yaml]: Done the change. DFU seems new - the full build does not do it either. I wonder if the full build should also have these flags.

Comment (Damian-Nordic) [reply .github/workflows/smoketest-nrfconnect.yaml]: In the full build, we build lighting-app with the DFU (by default), so we haven't enabled it for all-clusters-app. But we can consider changing that, too.

**Review by Damian-Nordic — APPROVED** [review]

**Pull request merged**